test/units/TEST-50-DISSECT.sysext.sh

```diff
@@ -999,6 +1063,55 @@ prepend_trap "chmod ${old_mode@Q} ${fake_root@Q}${hierarchy@Q}"
 chmod 0700 "$extension_data_dir"
 
 (! run_systemd_sysext "$fake_root" --mutable=yes merge)
+prepend_trap "rm -rf ${extension_data_dir@Q}"
+)
+
+( init_trap
+: "Check if merging an extension with matching ID succeeds"
+fake_root=${roots_dir:+"$roots_dir/matching-id"}
+hierarchy=/opt
+
+prepare_root "$fake_root" "$hierarchy"
+prepare_extension_image_with_matching_id "$fake_root" "$hierarchy"
+prepare_read_only_hierarchy "$fake_root" "$hierarchy"
+
+run_systemd_sysext "$fake_root" merge
+)
+
+( init_trap
+: "Check if merging an extension with non-matching ID fails"
+fake_root=${roots_dir:+"$roots_dir/non-matching-id"}
+hierarchy=/opt
+
+prepare_root "$fake_root" "$hierarchy"
+prepare_extension_image_with_non_matching_id "$fake_root" "$hierarchy"
+prepare_read_only_hierarchy "$fake_root" "$hierarchy"
+
+(! run_systemd_sysext "$fake_root" merge)
```

Comment:
missing prepend_trap ?

Reply:
The last test definitely doesn't cleanup after itself, I'm now getting a different err: `rm: cannot remove '/opt/preexisting-file-in-hierarchy': Read-only file system`

Pushed another attempt, but I don't really know what I'm doing with these tests anymore.

Reply:
works now.